src/http-spec.ts

```diff
@@ -35,10 +35,15 @@ export interface IHttpOperation extends INode {
   request?: IHttpOperationRequest;
   responses: IHttpOperationResponse[] & { 0: IHttpOperationResponse };
   servers?: IServer[];
+  callbacks?: IHttpCallbackOperation[];
   security?: HttpSecurityScheme[][];
   deprecated?: boolean;
 }
 
+export interface IHttpCallbackOperation extends IHttpOperation {
+  callbackName: string;
```

Comment:
I think this structure is too simplified. Let's look at this example:

```yaml
callbacks:
  myCallback:
    '{$request.query.queryUrl}':
      post:
        requestBody:
          description: Callback payload
          content: 
            'application/json':
              schema:
                $ref: '#/components/schemas/SomePayload'
        responses:
          '200':
            description: callback successfully processed
```
It seems like an operation could have multiple [Callback Objects](https://github.com/OAI/OpenAPI-Specification/blob/77e0e39708fcd0ae933fd7c6f960e32e861db4c6/versions/3.0.3.md#callbackObject), which each have their own name:

<img width="947" alt="image" src="https://user-images.githubusercontent.com/67381/67962032-c5f1f200-fbb9-11e9-8382-cbb211145d3b.png">

Inside this Callback Object it is another object, which could have multiple keys. This time, the key is a [runtime expression](https://github.com/OAI/OpenAPI-Specification/blob/77e0e39708fcd0ae933fd7c6f960e32e861db4c6/versions/3.0.3.md#runtimeExpression).

This is probably to allow multiple callbacks to be sent to multiple locations. Maybe one goes to the URL the user provided, and another goes to some central notifications API, or something like that. 

I think we probably need to make `IHttpCallbacks` which is an array and each has its own name, but each also has a dictionary of <string,PathItem> where string is the runtime expression, then the path item can contain the operation. 

Or something like that. I'm not a TS expert, but hopefully you get the idea?

Reply:
@philsturgeon 

It is exactly what we do:
```
callbacks:
  myCallback1:
    '{$request.body#/url}': ...
    '{$request.body#/url2}': ...
  myCallback2:
    '{$request.body#/ur3}': ...
    '{$request.body#/url4}': ...
```

is being transformed to:

```
const callbacks: IHttpCallbackOperation[] = [
{ name: 'myCallback1', path: '{$request.body#/url}' },
{ name: 'myCallback1', path: '{$request.body#/url2}' },
{ name: 'myCallback2', path: '{$request.body#/url3}' },
{ name: 'myCallback2', path: '{$request.body#/url4}' }
];
```

I'll add a test inside http-spec to emphasize that

Reply:
@philsturgeon Yeah so what we've been doing for `IHttpOperation` is fundamentally put them into an array instead of having an intermediate `IHttpPath` object containing these, and so the same thing happens for callbacks as well.